client/src/main/java/io/pravega/client/BatchClientFactory.java

```diff
@@ -103,4 +104,16 @@ static BatchClientFactory withScope(String scope, ClientConfig config) {
      * @return A list of segment range in between a start and end stream cut.
      */
     List<SegmentRange> getSegmentRangeBetweenStreamCuts(final StreamCut startStreamCut, final StreamCut endStreamCut);
+
+    /**
+     * Provides nearest streamcut in future depending on the distance and current streamcut.
+     * Depending on the requested distance per number of segments in the current streamcut, next offset for each segment is requested.
+     * If the current segment offset is at the tail of it, then the successor segment for it is being fetched.
+     * However, in case of scale down if offsets of all the segments participating in the scale down are at the tail then only call to get the next offset of their successor is made.
+     * @param startingStreamCut Starting streamcut
+     * @param approxDistanceToNextOffset approx distance to nextoffset in bytes
+     * @return A streamcut
+     * @throws SegmentTruncatedException If the segment is truncated or it does not exist.
+     */
+    StreamCut getNextStreamCut(final StreamCut startingStreamCut, long approxDistanceToNextOffset) throws SegmentTruncatedException;
```

Comment:
How should client react to `SegmentTruncatedException`? It looks like a common case if there is a streamcut which has been truncated, and then users want to proceed with the read.

Reply:
Removed

Reply:
How do we handle such truncated case internally? it would be better if there is some explanation in the javadoc, just like the scale-down case mentioned.

Reply:
> Restart reading from the curernt head

Is there a reason that such logic cannot be done internally in the client?

Reply:
(Added as you noted)
Conceivably we could. However I don't want an app to be unaware that it was attempting to read was deleted.